api/app/signals/apps/api/tests/test_public_signal_geography_endpoint.py

```diff
@@ -175,3 +172,42 @@ def test_get_group_by(self):
         data = response.json()
         self.assertEqual(1, len(data['features']))
         self.assertEqual(data['features'][0]['properties']['category']['name'], child_category.public_name)
+
+    def test_lon_lat_parameters(self):
+        """
+        Return the GEOJson containing all signals in the child category
+        """
+        parent_category = ParentCategoryFactory.create()
+        child_category = CategoryFactory.create(parent=parent_category, public_name='Public category for testing',
+                                                is_public_accessible=True)
+
+        signal = SignalFactoryValidLocation.create(category_assignment__category=child_category)
+        lon = signal.location.geometrie.x
+        lat = signal.location.geometrie.y
+
+        response = self.client.get(f'{self.geography_endpoint}/?maincategory_slug={parent_category.slug}'
+                                   f'&category_slug={child_category.slug}&lon={lon}&lat={lat}')
+
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(1, int(response.headers['X-Total-Count']))
+
+        data = response.json()
+        self.assertEqual(1, len(data['features']))
+
+        for feature in data['features']:
+            self.assertEqual(feature['properties']['category']['name'], child_category.public_name)
+
+    def test_missing_bbox_and_lonlat(self):
+        """
+        Validate that when neither bbox or lon/lat is filled in to return a non_field_error
+        """
+        parent_category = ParentCategoryFactory.create()
+        child_category = CategoryFactory.create(parent=parent_category, is_public_accessible=False)
+        response = self.client.get(f'{self.geography_endpoint}/?maincategory_slug={parent_category.slug}'
+                                   f'&category_slug={child_category.slug}')
+
+        data = response.json()
+        self.assertEqual(400, response.status_code)
+        self.assertIn('non_field_errors', data.keys())
+        self.assertEqual(1, len(data['non_field_errors']))
+        self.assertEqual('Either bbox or lon/lat must be filled in', data['non_field_errors'][0])
```

Comment:
Maybe also add a test for a lon without a lat and visa versa? Just to be complete.